src/w/index.ts

```diff
@@ -85,92 +100,125 @@ const wireReactor = <T>(fn: ($: SubToken) => T): WireReactor<T> => {
+const reactorUnsubscribe = (wR: WireReactor<X>): void => {
+  const unlinkFromSignal = (signal: WireSignal<X>) => signal.rS.delete(wR);
   // Skip newly created reactors since inner/rS/rP aren't yet defined
+  // TODO: Write a test for `runs` to check unexpected runs
   if (wR.runs) {
     wR.inner.forEach(reactorUnsubscribe);
-    wR.rS.forEach((signal) => signal.wR.delete(wR));
+    wR.sRS.forEach(unlinkFromSignal);
+    wR.sCS.forEach(unlinkFromSignal);
   }
-  wR.rS = new Set();
-  wR.rP = new Set();
+  wR.state = STATE_OFF;
   wR.inner = new Set();
+  // Drop all signals now that they have been unlinked
+  wR.sRS = new Set();
+  wR.sRP = new Set();
+  wR.sCS = new Set();
 };
 
-const reactorPause = (wR: WireReactor) => {
+const reactorPause = (wR: WireReactor<X>) => {
   wR.state = STATE_PAUSED;
   wR.inner.forEach(reactorPause);
 };
 
-const wireSignals = <T extends O>(obj: T): { [K in keyof T]: WireSignal<T[K]>; } => {
-  type V = T[keyof T];
+const wireSignals = <T>(obj: T): {
+  [K in keyof T]: WireSignal<T[K] extends WireReactor<infer R> ? R : T[K]>;
+} => {
   Object.keys(obj).forEach((k) => {
-    let saved = obj[k];
-    let reactorForToken: WireReactor | undefined;
-    // Batch the identifier since key k will be unique
-    const id = `wS#${signalId}(${k})`;
-    const wS = { [id](...args: (V | SubToken)[]) {
-      // Case: Read-Pass
-      if (!args.length) {
+    let saved: unknown;
+    let read: WireReactor<X> | boolean | undefined; // Multi-use temp variable
+    const id = `wS:${signalId++}{${k}}`;
+    const wS = { [id](...args: (SubToken | unknown)[]) {
+      // Case: Read-Subscribe
+      // eslint-disable-next-line no-cond-assign
+      if (read = !args.length) {
         if (activeReactor) {
-          if (activeReactor.rS.has(wS)) {
-            throw new Error(`Mixed rS/rP ${wS.name}`);
+          if (activeReactor.sRS.has(wS)) {
+            throw new Error(`Mixed sRS/sRP ${wS.name}`);
           }
-          activeReactor.rP.add(wS);
+          activeReactor.sRP.add(wS);
         }
       }
-      // Case: Read-Sub; could be any reactor not necessarily `reactorActive`
+      // Case: Read-Pass; could be any reactor not necessarily `activeReactor`
       // eslint-disable-next-line no-cond-assign
-      else if (reactorForToken = reactorTokenMap.get(args[0] as SubToken)) {
-        if (reactorForToken.rP.has(wS)) {
-          throw new Error(`Mixed rP/rS ${wS.name}`);
+      else if (read = reactorTokenMap.get(args[0] as SubToken)) {
+        type R = WireReactor<X>;
+        if ((read as R).sRP.has(wS)) {
+          throw new Error(`Mixed sRP/sRS ${wS.name}`);
         }
-        reactorForToken.rS.add(wS);
-        wS.wR.add(reactorForToken);
+        (read as R).sRS.add(wS);
+        wS.rS.add((read as R));
+        // Subscribing to a computed-signal also links cR's subscribed signals
+        wS.cR && wS.cR.sRS.forEach((s) => {
+          // Link to sXS not sRS. This way the "Mixed A/B" errors keep working
```

Comment:
Need to write tests for this 😅

Reply:
This is very very true now. Might not work correctly.

Reply:
OH! I meant "This was errors keep working" as it the errors **don't throw** despite the linking. The linking is for debugging at this point. It's to see that there's a connection, but neither error branch uses the inherited-signals (`core.sCS`) or the signal's reactor list (`signal.rS`). Instead, `core.sRP` and `core.sRS` are checked.